src/convertToObject.js

```diff
@@ -1,12 +1,68 @@
 'use strict';
 
-/**
- * @param {string} sourceString
- *
- * @return {object}
- */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString === '') {
+    return {};
+  }
+
+  const firstObrabotka = sourceString;
+
+  const bigArr = firstObrabotka.split(';');
+
+  const arrFinal = bigArr.filter((str) => str.trim() !== '');
+
+  let smallArr;
+  const result = {};
+
+  for (let i = 0; i < arrFinal.length; i++) {
+    smallArr = arrFinal[i].split(':');
+
+    const key = smallArr[0].trim();
+    const value = smallArr[1].trim();
+
+    result[key] = value;
+  }
```

Comment:
Instead of using a for loop, you can use array iteration methods like 'map' to transform each style declaration into an object and then use 'Object.assign' or the spread operator to merge them into a single object. This would be more in line with functional programming practices.